src/Umbraco.Web.UI.Client/src/views/components/tree/umb-tree.html

```diff
@@ -13,26 +13,28 @@ <h5>
                 </a>
             </h5>
 
-            <umb-button-ellipsis
-                element="tree-item-options"
-                action="options(tree.root, $event)"
-                text="Open context node for {{tree.name}}"
-                state="hidden"
-                class="umb-options"
-                ng-hide="tree.root.isContainer || !tree.root.menuUrl"
-                ng-swipe-right="options(tree.root, $event)">
+            <umb-button-ellipsis element="tree-item-options"
+                                 action="options(tree.root, $event)"
+                                 text="Open context node for {{tree.name}}"
+                                 state="hidden"
+                                 class="umb-options"
+                                 ng-hide="tree.root.isContainer || !tree.root.menuUrl"
+                                 ng-swipe-right="options(tree.root, $event)">
             </umb-button-ellipsis>
 
         </div>
-        <umb-tree-item class="umb-animated"
-                       ng-repeat="child in tree.root.children"
-                       enablelistviewexpand="{{enablelistviewexpand}}"
-                       node="child"
-                       current-node="currentNode"
-                       tree="this"
-                       is-dialog="isdialog"
-                       section="{{section}}">
-        </umb-tree-item>
+
+        <ul>
```

Comment:
It should probably also include `ng-if="tree.root.children.length > 0` to ensure it doesn't render an empty `<ul>` without `<li>` elements.

Reply:
@bjarnef I agree on this, I have made a condition on this, on both ul elements, to ensurce is does not render if empty